pkg/bloomcompactor/bloomcompactor.go

```diff
@@ -354,46 +353,53 @@ func (c *Compactor) compactTenant(ctx context.Context, logger log.Logger, sc sto
 	NGramSkip := c.limits.BloomNGramSkip(tenant)
 	bt, _ := v1.NewBloomTokenizer(c.reg, NGramLength, NGramSkip)
 
-	// TODO: Use ForEachConcurrent?
 	errs := multierror.New()
 	if err := sc.indexShipper.ForEach(ctx, tableName, tenant, func(isMultiTenantIndex bool, idx shipperindex.Index) error {
-		if isMultiTenantIndex {
+		if isMultiTenantIndex { // TODO: handle multitenant tables
 			return fmt.Errorf("unexpected multi-tenant")
 		}
 
+		var indices []Index
 		// TODO: Make these casts safely
 		if err := idx.(*tsdb.TSDBFile).Index.(*tsdb.TSDBIndex).ForSeries(
 			ctx, nil,
 			0, math.MaxInt64, // TODO: Replace with MaxLookBackPeriod
 			func(labels labels.Labels, fingerprint model.Fingerprint, chksMetas []tsdbindex.ChunkMeta) {
-				job := NewJob(tenant, tableName, idx.Path(), fingerprint, labels, chksMetas)
-				jobLogger := log.With(logger, "job", job.String())
+				ownsFingerprint, err := c.sharding.OwnsFp(tenant, uint64(fingerprint))
+				fpLogger := log.With(logger, "fp", tableName+"_"+tenant+"_"+fingerprint.String())
 
-				ownsJob, err := c.sharding.OwnsJob(job)
 				if err != nil {
-					c.metrics.compactionRunUnownedJobs.Inc()
-					level.Error(jobLogger).Log("msg", "failed to check if compactor owns job", "err", err)
+					c.metrics.compactionRunSkippedFp.Inc()
+					level.Error(fpLogger).Log("msg", "failed to check if compactor owns fp", "err", err)
 					errs.Add(err)
 					return
 				}
-				if !ownsJob {
-					c.metrics.compactionRunUnownedJobs.Inc()
-					level.Debug(jobLogger).Log("msg", "skipping job because it is not owned by this shard")
+				if !ownsFingerprint {
+					c.metrics.compactionRunSkippedFp.Inc()
+					level.Debug(fpLogger).Log("msg", "skipping fp because it is not owned by this shard")
```

Comment:
I believe this message will be too noisy because if we have 12 compactors,  each compactor will report all fingerprints from the rest 11 ranges ... let's say if we have 100k streams, each compactor will report 91.6k fingerprints

Reply:
agreed, Salva had the same concern when adding. I removed.